dask/dataframe/io/parquet.py

```diff
@@ -174,8 +175,17 @@ def _read_pyarrow(fs, paths, file_opener, columns=None, filters=None,
 
     dataset = api.ParquetDataset(paths)
     schema = dataset.schema.to_arrow_schema()
+    has_pandas_metadata = schema.metadata is not None and b'pandas' in schema.metadata
     task_name = 'read-parquet-' + tokenize(dataset, columns)
 
+    if index is False:
+        index_col = None
+    elif index is None and has_pandas_metadata:
+        pandas_metadata = json.loads(schema.metadata[b'pandas'].decode('utf8'))
+        index_col = pandas_metadata.get('index_columns', None)
+    else:
+        index_col = index
```

Comment:
Can any of this metadata wrangling be pushed down into pyarrow?

Reply:
The retrieval of the index column should be definitely pushed down to `pyarrow`. Currently there is no such functionality on the `ParquetDataset`. I can add that for future releases. This PR mainly covers the parts where we can satisfy the tests without developing new features in `pyarrow`.